docs/reference/hooks/interchain-gas.mdx

```diff
@@ -31,28 +29,85 @@ struct StandardHookMetadata {
+- `msgValue`: Amount of native token sent with the message.
+- `gasLimit`: Gas limit for the `handle` function on the destination chain. Ensure this matches your simulation results.
+- `refundAddress`: Address to refund unused gas payments.
+
+To encode this metadata, use the `StandardHookMetadata.formatMetadata` library function. Direct encoding of the struct with `abi.encodePacked` is not supported in Solidity.
+
+#### Example Usage
+
+<Tabs groupId="lang">
+<TabItem value="sol" label="Solidity">
 
 ```solidity
-  bytes memory hookMetadata = StandardHookMetadata.formatMetadata(
-    0,                   // ETH message value
-    200000,              // 200k gas limit
-    address(this),       // refund address
-    bytes("")            // custom metadata
-  });
+// Example: Encoding metadata using StandardHookMetadata
+bytes memory metadata = StandardHookMetadata.formatMetadata(
+    0,               // ETH message value
+    200000,          // Custom gas limit
+    address(this),   // Refund address
+    bytes("")        // Optional custom metadata
+);
 ```
 
+### Determine and Override the Gas Limit
+
+1. Simulate and Benchmark Gas Usage:
+
+- Use tools like Tenderly or Foundry to simulate your transaction and measure gas consumption.
+- If gas usage exceeds `50,000` gas, calculate an appropriate `gasLimit` and update your metadata.
+
+2. Update Your Metadata:
+
+- Calculate the required `gasLimit` based on the simulation.
+- Pass the updated `gasLimit` in your metadata to prevent transaction reverts.
+
 </TabItem>
 </Tabs>
 
-## Gas Oracles
+## Destination Gas Config
+
+For each remote domain, the InterchainGasPaymaster sets the domain gas config.
+
+```solidity
+struct DomainGasConfig {
+    IGasOracle gasOracle;
+    uint96 gasOverhead;
+}
+```
 
-The interchain gas payment requirement is calculated using oraclized gas price and exchange rates between supported origin and destination chains.
+### Gas overhead
+
+The gas overhead is set as part of the destination gas configuration. This corresponds to the operational cost of processing a message on the destination chain.
 
 :::info
-Exchange rates and gas prices are up to the relayer to decide. A spread may be charged to account for drift and operating costs.
+
+- You should ensure the `gasOverhead` sufficiently covers the range of ISMs on your destination chain.
+- As you can configure different ISMs for different message types, you may have different gas overheads for each ISM's `verify` function.
+
 :::
 
+## Gas Oracles
+
+Interchain Gas Payment requirements are calculated using oraclized gas prices and exchange rates between the origin and destination chains.
+
+IGP contracts may be configured with gas oracles, which are responsible for tracking remote token gas prices and exchange rates. This allows the `quoteGasPayment` function to provide an accurate quote for the price of gas on a remote chain, denominated in the local chain's native token.
```

Comment:
see my comments below about `quoteGasPayment`
we really only want devs to go through `quoteDispatch` on the mailbox at this point